src/main/scala/com/monovore/decline/Argument.scala

```diff
@@ -5,7 +5,7 @@ import java.nio.file.{InvalidPathException, Path, Paths}
 
 import cats.data.{Validated, ValidatedNel}
 
-trait Argument[A] {
+trait Argument[A] { self =>
```

Comment:
I had `map` on here briefly, IIRC, but ended up dropping it... didn't save much code, and I ended up forgetting to update the `defaultMetavar`.

On the fence about sealing -- I kind of like the fact that folks can create an `Argument[KafkaTopic]` or whatever and have things work out. What kind of misuse are you picturing here?